### ISSUE TITLE
tf.signal.ifft returns results with different dtypes for subsequent calls with the same parameter

### ISSUE THREAD

**Issue by bchetioui** [tensorflow/tensorflow#43115]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 10.4
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v1.12.1-40510-ga32c74ae8f 2.4.0-dev20200829
- Python version: Python 3.8.3
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

When calling `tf.signal.ifft` with an operand with a `complex128` dtype several times, the first call returns a `complex128` array as expected, but all subsequent calls return a `complex64` array.

**Describe the expected behavior**

`tf.signal.ifft` should always return an array with the same type as its input per the documentation; more importantly, it should be consistent across calls.

**Standalone code to reproduce the issue**

```python
import numpy as np
import tensorflow as tf

operand = np.ones((2, 2), dtype=np.complex128)
print(tf.signal.ifft(operand).dtype)
print(tf.signal.ifft(operand).dtype)
```

**Comment by bhack** [comment]
For me It works with tf-nightly and:
```
print(tf.signal.ifft([operand]).dtype)
print(tf.signal.ifft([operand]).dtype)
```

**Comment by TuringEmmy** [comment]
for me it is:
**2020-09-11 11:01:58.434235**: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
**2020-09-11 11:01:58.458559**: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7f80984b9190 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
**2020-09-11 11:01:58.458578**: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
tf.complex64



you need rebuild TensorFlow with the appropriate compiler flags. Because this TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA

**Comment by bhack** [comment]
@TuringEmmy try with https://pypi.org/project/tf-nightly-cpu/

**Comment by Saduf2019** [comment]
@TuringEmmy 
Please update as per above comment.

**Comment by SIben** [comment]
@Saduf2019
I assume you mean me, as @TuringEmmy is not the original author of the issue.
I am using the TF nightly build `2.4.0-dev20200829`; isn't that (one of) the most recent builds? Nevertheless, I upgraded to `v1.12.1-41432-g3e446295d8 2.4.0-dev20200912`, the latest version I was able to get through pip, and the issue is still here.

> For me It works with tf-nightly and:
> 
> ```
> print(tf.signal.ifft([operand]).dtype)
> print(tf.signal.ifft([operand]).dtype)
> ```

@bhack This indeed works, but the behavior is still different across calls without the brackets. Any insight why?
